src/irmin-containers/linked_log.mli

```diff
@@ -0,0 +1,83 @@
+(*
+ * Copyright (c) 2020 KC Sivaramakrishnan <kc@kcsrk.info>
+ * Copyright (c) 2020 Anirudh Sunder Raj <anirudh6626@gmail.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+(** [Linked_log] is a linked list implementation of the log. In this, two copies
+    of the log share their common predecessor. The type of the log entries, the
+    method to obtain timestamp, a hashing method and a content addressable store
+    maker are provided by the user. *)
+
+(** Signature of [Linked_log] *)
+module type S = sig
+  module Store : Irmin.S
+  (** Store for the log. All store related operations like branching, cloning,
+      merging, etc are done through this module. *)
+
+  type value
+  (** Type of log entry *)
+
+  type cursor
+  (** Type of cursor. Cursor is like a marker from which a certain number of
+      entries can be read *)
+
+  val append :
+    info:Irmin.Info.f -> Store.t -> path:Store.key -> value -> unit Lwt.t
+  (** Append an entry to the log *)
+
+  val get_cursor : Store.t -> path:Store.key -> cursor option Lwt.t
+  (** Get the cursor *)
```

Comment:
It's not clear to me from reading this signature why a `cursor` should depend on a particular `path`, or why it's _the_ cursor for that `path`. Perhaps something like:

> Create a new cursor over log entries at the given path.

I also wonder if it should just return `cursor Lwt.t` and simply not read any entries if no values have been `append`-ed to that key.

Reply:
Your suggestion is a pretty accurate description of `get_cursor`. I will update the documentation. Also, returning only `cursor Lwt.t` sounds like a good idea.